### DISCUSSION TITLE
Why can't DocQuery extract multiple lineitems?

### DISCUSSION THREAD

**Discussion by ankrgyl** [impira/docquery#33] [Q&A]
Moving the conversation from #26 over to a new discussion (CC @izdi):

@ankrgyl yes. Playing with it looks promising!

One question, tho', if you ask, "What is the description of all items?" it returns only the first one or the last one depending on the framing of the question. Why do you think that might happen?

```
$ docquery scan "What is the description of all line items?" https://templates.invoicehome.com/invoice-template-us-neat-750px.png

image-classification is already registered. Overwriting pipeline for task image-classification...
2022-09-14 17:17:20,105 INFO: Loading https://templates.invoicehome.com/invoice-template-us-neat-750px.png
2022-09-14 17:17:20,105 INFO: Done loading files. Loading pipeline...2022-09-14 17:17:28,125 INFO: Ready to start evaluating!

https://templates.invoicehome.com/invoice-template-us-neat-750px.png What is the description of all line items?: Front and rear brake cables

(base) izdi at ~
$ docquery scan "What is the description of all items?" https://templates.invoicehome.com/invoice-template-us-neat-750px.png

image-classification is already registered. Overwriting pipeline for task image-classification...
2022-09-14 17:18:23,226 INFO: Loading https://templates.invoicehome.com/invoice-template-us-neat-750px.png
2022-09-14 17:18:23,226 INFO: Done loading files. Loading pipeline...
2022-09-14 17:18:30,372 INFO: Ready to start evaluating!

https://templates.invoicehome.com/invoice-template-us-neat-750px.png What is the description of all items?: Front and rear brake cables

(base) izdi at ~
$ docquery scan "What is the description of QTY 1 2 3?" https://templates.invoicehome.com/invoice-template-us-neat-750px.png

image-classification is already registered. Overwriting pipeline for task image-classification...
2022-09-14 17:18:56,499 INFO: Loading https://templates.invoicehome.com/invoice-template-us-neat-750px.png
2022-09-14 17:18:56,499 INFO: Done loading files. Loading pipeline...
2022-09-14 17:19:04,022 INFO: Ready to start evaluating!

https://templates.invoicehome.com/invoice-template-us-neat-750px.png What is the description of QTY 1 2 3?: Labor Shrs
```

**Comment by ankrgyl** [comment]
@izdi this is a great question. The model currently is not "framed" to support extracting multiple lineitems. This is a limitation _both_ in the data used to train it as well as the output of the model itself (specifically how the logits are used).

We are actually working right now on a new iteration of the model that supports extracting lineitems. Our commercial product (http://impira.com/) already supports them if you need it sooner.

**Reply by absurdity9** [reply]
When is the new iteration released?
